### ISSUE TITLE
Create a release process

### ISSUE THREAD

**Issue by ColinEberhardt** [facebook/css-layout#101]
This has turned into a really popular library, which I have seen used in  a number of interesting ways - :+1: - however, it currently seems to lack a release process.

I am interested in this because I am contributor to a library, http://d3fc.io/, which depends on css-layout. As a result, I am primarily interested in the JavaScript version of css-layout.

I'd really like to give our users options for obtaining dependencies (d3 and css-layout) other than npm, simply because there will be many front-end devs who do not use npm.

What I'd really love to see is the following:

 + minified JavaScript files
 + more frequent releases to npm
 + GitHub releases with associated source-code
 + the JavaScript code available via CDNJS

Minification could be achieved by adding a very simple grunt build (you already have test and pre-tests steps in package.json, having them in grunt would be cleaner in my opinion).

The release process could all be managed via travis, see our deploy configuration here:

https://github.com/ScottLogic/d3fc/blob/master/.travis.yml

See: https://github.com/ScottLogic/d3fc/issues/467

**Comment by vjeux** [comment]
That's a good idea. Are you interested in doing it? If yes, i can give you commit access and work with you on it :)

**Comment by jaredly** [comment]
@ColinEberhardt that layout component is a neat application of css-layout to svg :) sounds really useful

**Comment by ColinEberhardt** [comment]
Thanks @jaredly :-)

Yes, @vjeux - happy to implement this. I'd like to split it into steps:

1) Create a grunt build that tests, minifies and creates the release distributions
2) Hook (2) up to Travis
3) Update the Travis config to perform the releases

I'll start work on this over the next couple of weeks.

**Comment by vjeux** [comment]
Sounds like a good plan :)

**Comment by ColinEberhardt** [comment]
@vjeux - so after battling with Travis for a few hours ... I gave up! They have been making changes to the way npm integration works and it looks like I am not the only one struggling with npm deploy automation.

Anyhow, I have instead opted for a documented release process. It's only 4 steps, so not too bad:

https://github.com/facebook/css-layout/tree/master/dist

I've tested the process via a patch release, 0.0.6 - now that it is all working are you happy to move into semver 'proper' and have a 1.0.0 release?
